libp2p/network/url_conversion.go

```diff
@@ -127,7 +127,30 @@ func getPeerIP(hostname string) (net.IP, error) {
 		if peerIP == nil {
 			continue
 		}
+		// TECHDEBT: remove this log line once direction is clearer
+		// on supporting multiple network addresses per peer.
+		if len(addrs) > 1 {
+			logger.Global.Warn().
+				Array("resolved", newStringLogArrayMarshaler(addrs)).
+				IPAddr("using", peerIP)
+		}
 		return peerIP, nil
 	}
 	return nil, newResolvePeerIPErr(hostname, err)
 }
+
+type stringLogArrayMarshaler struct {
+	strs []string
+}
+
+func (marshaler stringLogArrayMarshaler) MarshalZerologArray(arr *zerolog.Array) {
```

Comment:
Where is this being used?

Reply:
This is called by `zerolog` and is required by the `zerolog.LogArrayMarshaler` interface, which is the type of the second argument to [`zerolog.Evt#Array`](https://pkg.go.dev/github.com/rs/zerolog@v1.27.0#section-readme). 

I don't have a strong opinions on doing it this way, this just seemed to me like the most appropriate way to convey the information available via our structured logging. I've simplified as you suggested and added comments so nobody else has to ask the same question. :sweat_smile: